scripts/sysman/memState.yml

```diff
@@ -0,0 +1,71 @@
+#
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+#
+# See YaML.md for syntax definition
+#
+--- #--------------------------------------------------------------------------
+type: header
+desc: "Intel $OneApi Level-Zero Sysman Extension APIs for Memory State"
+version: "1.8"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Memory State Extension Name"
+version: "1.8"
+name: $S_MEMORY_STATE_EXP_NAME
+value: '"$XS_extension_memory_state"'
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Memory State Extension Version(s)"
+version: "1.8"
+name: $s_memory_state_exp_version_t
+etors:
+    - name: "1_0"
+      value: "$X_MAKE_VERSION( 1, 0 )"
+      desc: "version 1.0"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Extension properties related to Memory Modules"
+version: "1.8"
+class: $sMemory
+name: $s_mem_exp_properties_t
+base: $s_base_properties_t
+members:
+    - type: uint32_t
+      name: "countOfVirtualFunctionInstance"
+      desc: |
+            [out] Number of Virtual Function(VF) instances associated with memory module to monitor the utilization of hardware across all VFs from a Physical Function (PF) instance.
+            This count represents the number of VF instances that are actively using the resource represented by the memory module handle.
+details:
+    - "This structure may be passed to $sMemoryGetProperties by having the pNext member of $s_mem_properties_t point at this struct."
+    - "Used for SRIOV per VF device utilization by $s_mem_type_t"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Get memory state for Physical Function (PF) and each Virtual Function (VF) associated with memory module."
+version: "1.8"
+class: $sMemory
+name: GetStateExp
```

Comment:
can we name this less generic? this is state of Physical and Virtual functions, so maybe something like zesMemoryGetVirtualAndPhyscalFunctionsStateExp, but maybe that's too long.

Reply:
no, this would be zesMemoryGetStateExp, so intention is to gather memory state (alloc/free counters) for VF and PF

Reply:
so does that mean that [zesMemoryGetState](https://spec.oneapi.io/level-zero/latest/sysman/api.html?highlight=zesmemorygetstate#_CPPv417zesMemoryGetState16zes_mem_handle_tP15zes_mem_state_t) would be deprecated? if it is for all memory, do we need to mention about VF and PF?

Reply:
@aravindksg Couldnt we extend existing zesMemroyGetState by adding a new field in zes_structure_type_t. As i feel that current name of API zesMemoryGetState is good. All other names sounds too much elaborative.

Reply:
@jandres742 no `zesMemoryGetState` is still valid and might return valid info for environments when info can be retrieved. But we need separate API for gathering virtualized memory state (similar to how we do for engine stats via `zesEngineGetActivityExt`) from PF/VF domains. hence the proposal here. 

@jitendrasharma1989 each structure only has one struct type. don't think we can define new struct type and use with existing structure. 

As mentioned below, thinking of going with `zesMemGetVirtualizedStateExp` (which is probably not too long)